### PR TITLE
Use server sent events to maintain a client side view of the keyspace schemas #554

### PR DESCRIPTION
**Pull request by tonkolviktor**
Hey,

Thanks for the lot of work put into this project.
We needed the shcema update feature, so i implemented based on our requirements.

Hope it helps and can be merged (or at least starts a conversation :))

### CODE REVIEW

File: events.go [hunk 1]
```diff
@@ -107,6 +107,14 @@ func (s *Session) handleEvent(framer *framer) {
 }
 
 func (s *Session) handleSchemaEvent(frames []frame) {
+	for _, frame := range frames {
```
Comment (Zariel) [inline events.go]: It looks like there are cases where the sessions `schemaDescriber` is nil, we should check and not clear metadata.

File: events.go [hunk 2]
```diff
@@ -107,6 +107,9 @@ func (s *Session) handleEvent(framer *framer) {
 }
 
 func (s *Session) handleSchemaEvent(frames []frame) {
+	if s.schemaDescriber == nil {
```
Comment (tonkolviktor) [inline events.go]: thanks good point, fixed

**Comment by Zariel** [comment]
Thanks, if not already can you add yourself to the AUTHORS file and I will merge.

**Comment by tonkolviktor** [comment]
done, thanks

**Comment by Zariel** [comment]
Thank you very much for the contribution!

**Pull request merged**